src/ucs/stats/stats.c

```diff
@@ -117,38 +117,49 @@ static void ucs_stats_clean_node(ucs_stats_node_t *node) {
     ucs_list_del(&node->type_list);
 }
 
+static void ucs_stats_free_class(ucs_stats_class_t *cls)
+{
+    unsigned i;
+
+    for (i = 0; i < cls->num_counters; i++) {
+        ucs_free((void*)cls->counter_names[i]);
+    }
+
+    ucs_free((void*)cls->name);
+    ucs_free(cls);
+}
+
 static ucs_stats_class_t *ucs_stats_dup_class(ucs_stats_class_t *cls)
 {
     ucs_stats_class_t *dup;
 
-    dup = ucs_malloc(sizeof(*cls) + sizeof(*cls->counter_names) * cls->num_counters,
-                     "ucs_stats_class_t");
-    ucs_assertv_always(dup != NULL, "failed to allocate statistics class");
+    dup = ucs_calloc(1, sizeof(*cls) + sizeof(*cls->counter_names) * cls->num_counters,
+                     "ucs_stats_class_dup");
+    if (!dup) {
+        ucs_warn("failed to allocate statistics class");
+        return NULL;
```

Comment:
goto err;
(add label err: in the end, so all function returns will go thru same place)

Reply:
added label